framework/src/org/apache/cordova/PluginManager.java

```diff
@@ -381,15 +374,16 @@ public void onReset() {
     }
 
     Uri remapUri(Uri uri) {
+        Uri remap = uri;
         for (CordovaPlugin plugin : this.pluginMap.values()) {
             if (plugin != null) {
-                Uri ret = plugin.remapUri(uri);
+                Uri ret = plugin.remapUri(remap);
                 if (ret != null) {
-                    return ret;
+                    remap = ret;
                 }
             }
         }
-        return null;
+        return uri.equals(remap) ? null : remap;
```

Comment:
I suppose this returns null if no plugins ends up remapping the uri. However, a comment as to why this is being done would be great !